src/app/loginComponents/onboarding/downloadcliclient/downloadcliclient.component.ts

```diff
@@ -38,20 +39,29 @@ export class DownloadCLIClientComponent implements OnInit {
       (resultFromApi: Metadata) => {
         apiVersion = resultFromApi.version;
         this.dockstoreVersion = `${apiVersion}`;
-        this.downloadCli = `https://github.com/dockstore/dockstore/releases/download/${apiVersion}/dockstore`;
-        this.metadataService
-          .getRunnerDependencies(apiVersion, '3', 'cwltool', 'json')
-          .pipe(finalize(() => this.generateMarkdown()))
-          .subscribe(
-            (json: any) => {
-              if (json) {
-                this.cwltoolVersion = json.cwltool;
-              }
-            },
-            (err) => {
-              console.log('Unable to retrieve requirements.txt file.');
+        this.metadataService.getCliVersion().subscribe(
+          (json: CLIInfo) => {
+            if (json) {
+              this.downloadCli = json.cliLatestDockstoreScriptDownloadUrl;
             }
-          );
+            this.metadataService
+              .getRunnerDependencies(apiVersion, '3', 'cwltool', 'json')
+              .pipe(finalize(() => this.generateMarkdown()))
+              .subscribe(
+                (json: any) => {
+                  if (json) {
+                    this.cwltoolVersion = json.cwltool;
+                  }
+                },
+                (err) => {
+                  console.log('Unable to retrieve requirements.txt file.');
```

Comment:
Depending upon what we want to happen when we can't retrieve the necessary information, `ngOnInit` might need a few tweaks.  Do we always generate the markdown, for every error?  Use some rationale defaults in lieu of actual values?  Conditionally render the markdown to omit the parts we don't know?  Or, maybe failure is rare enough in this case that the particular behavior doesn't really matter, as long as it's reasonable?

Reply:
Good points, and your comment made me notice, we shouldn't be doing `console.log`, we should be surfacing the error.